cmd/juju/application/deployer/deployer.go

```diff
@@ -104,7 +105,11 @@ func (d *factory) GetDeployer(ctx context.Context, cfg DeployerConfig, getter Mo
 		return nil, errors.Trace(fileStatErr)
 	}
 
-	if charm.IsValidLocalCharmOrBundlePath(d.charmOrBundle) || isLocalSchema(d.charmOrBundle) {
+	if isLocalSchema(d.charmOrBundle) && charm.IsValidLocalCharmOrBundlePath(d.charmOrBundle[6:]) {
```

Comment:
This is a little more complex than it looks, since we call IsValidLocalCharmOrBundlePath with different params

But you're right, there is some redundancy here. CutPrefix can save us here, given it's idempotency (or close enough)